lambdas/db-provision-user-database/src/index.ts

```diff
@@ -0,0 +1,75 @@
+import Knex from 'knex';
+import AWS from 'aws-sdk';
+
+export interface HandlerEvent {
+  rootLoginSecret: string,
+  userLoginSecret: string,
+  prefix: string,
+  dbPassword: string,
+  engine: string,
+  dbClusterIdentifier: string,
+  env?: NodeJS.ProcessEnv,
+}
+
+export interface getSecretValue extends
+  AWS.Request<AWS.SecretsManager.GetSecretValueResponse, AWS.AWSError> {
+  SecretString: string
+}
+
+const getConnectionConfig = async (SecretId: string): Promise<Knex.PgConnectionConfig> => {
+  const secretsManager = new AWS.SecretsManager();
+  const response = await secretsManager.getSecretValue(
+    { SecretId } as AWS.SecretsManager.GetSecretValueRequest
+  ).promise();
+  if (response.SecretString === undefined) {
+    throw new Error('Database credentials are undefined!');
+  }
+  const dbAccessMeta = JSON.parse(response.SecretString);
+  return {
+    host: dbAccessMeta.host,
+    user: dbAccessMeta.username,
+    password: dbAccessMeta.password,
+  };
+};
+
+export const handler = async (event: HandlerEvent): Promise<void> => {
+  let knex;
+  try {
+    const secretsManager = new AWS.SecretsManager();
+    const config = await getConnectionConfig(event.rootLoginSecret);
+    knex = Knex({
+      client: 'pg',
+      connection: config,
+      acquireConnectionTimeout: 60000,
+    });
+    const dbUser = event.prefix.replace('-', '_');
+
+    const tableExists = await knex.select(1).as('result')
+      .from('pg_database').where('datname', `${dbUser}_db`);
+
+    if (tableExists.length === 0) {
+      await knex.raw(`create user ${dbUser} with encrypted password '${event.dbPassword}'`);
```

Comment:
There are probably some questions here regarding if we want a more complicated schema/permissions structure, however that's probably an additional ticket.   Comments welcome.

Reply:
An unrelated and possibly dumb question: do we need to be concerned about injection here when using `knex.raw`, since it's not escaping the input parameters?

Reply:
Injection is always a serious concern.     To some degree this lambda is protected by IAM authentication/user setup, but there's always a possibility someone does something that allows this lambda to get run with a bad payload.   Nice catch. 

  I think we're going to need to iterate on this setup, regardless, but adding a fairly restrictive regexp sanitizer for now.

Reply:
f12a2066b

Reply:
Do we want to validate the database password as well?

Reply:
Yes indeed.  c61fb224f.     I think to some degree, the amount of concern greatly depends on if this winds up being a user deliverable - but given comments here, I'm going to add unit tests around it.

Reply:
Hate to keep being pedantic here, but forcing the password to only be alphanumeric is probably not great for security. Sorry.

Reply:
Not entirely pedantic - to some degree the question is if a 60char alphanum versus a user putting in a 10char special char string here is meaningfully different for the purpose here.    If it weren't for IAM API restrictions as relates to scaling we wouldn't be *using* them at all.  

That said, it's not a huge issue to use a more permissive regexp, so long as we're careful about it.     I'll see if someone has implemented something more useful than my arbitrary regexp sanitization for PSQL.

Reply:
Also if you don't like that, I'd expect you *really* don't like 

https://github.com/nasa/cumulus/pull/1826/files#diff-e7e41536cc3b3b76b76cdd76fefc9a23R11

Reply:
Ok, I've added c14006ba6 (see comment about unit tests).   The real problem here is those queries cannot be parameterized... and really the right thing to do is make sure this lambda (and therefore access to the root db creds) are never run/accessed by anyone without deploy credentials (and therefore access to just change the db pw anyway).

Reply:
tl;dr - move this module to `example/` to reinforce the fact that it should not be released to users